File: src/main/java/org/jesperancinha/airports/containers/AirportContainer.java
package org.jesperancinha.airports.containers;

import com.fasterxml.jackson.databind.MappingIterator;
import com.fasterxml.jackson.dataformat.csv.CsvMapper;
import com.fasterxml.jackson.dataformat.csv.CsvSchema;
import org.jesperancinha.airports.pojos.Airport;
import lombok.Getter;

import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by joaofilipesabinoesperancinha on 01-08-16.
 */
@Getter
public class AirportContainer {

    private final CsvMapper mapper = new CsvMapper();
    private final CsvSchema schema = CsvSchema.builder()
            .setColumnSeparator(',')
            .addColumn("id")
            .addColumn("ident")
            .addColumn("type")
            .addColumn("name")
            .addColumn("latitudeDeg")
            .addColumn("longitudeDeg")
            .addColumn("elevationFt")
            .addColumn("continent")
            .addColumn("isoCountry")
            .addColumn("isoRegion")
            .addColumn("municipality")
            .addColumn("scheduledService")
            .addColumn("gpsCode")
            .addColumn("iataCode")
            .addColumn("localCode")
            .addColumn("homeLink")
            .addColumn("wikipediaLink")
            .addColumn("keywords")
            .build();


    private List<Airport> airports;

    public AirportContainer() throws IOException {
        airports = new ArrayList<>();
        final InputStream aiportStream = getClass().getResourceAsStream("/airports.csv");
        final MappingIterator<Airport> it = mapper
                .reader(Airport.class)
                .with(schema)
                .readValues(new InputStreamReader(aiportStream));
        airports = it.readAll();
    }
}

File: src/main/kotlin/org/jesperancinha/airports/containers/AirportContainer.kt
package org.jesperancinha.airports.containers

import com.fasterxml.jackson.dataformat.csv.CsvMapper
import com.fasterxml.jackson.dataformat.csv.CsvSchema
import org.jesperancinha.airports.pojos.Airport
import java.io.InputStreamReader

/**
 * Created by joaofilipesabinoesperancinha on 01-08-16.
 */
class AirportContainer {
    private val mapper: CsvMapper by lazy { CsvMapper() }
    private val schema: CsvSchema? by lazy {
        CsvSchema.builder()
            .setColumnSeparator(',')
            .addColumn("id")
            .addColumn("ident")
            .addColumn("type")
            .addColumn("name")
            .addColumn("latitudeDeg")
            .addColumn("longitudeDeg")
            .addColumn("elevationFt")
            .addColumn("continent")
            .addColumn("isoCountry")
            .addColumn("isoRegion")
            .addColumn("municipality")
            .addColumn("scheduledService")
            .addColumn("gpsCode")
            .addColumn("iataCode")
            .addColumn("localCode")
            .addColumn("homeLink")
            .addColumn("wikipediaLink")
            .addColumn("keywords")
            .build()
    }
    val airports: List<Airport> by lazy {
        mapper.reader(Airport::class.java)
            .with(schema)
            .readValues<Airport>(
                AirportContainer::class.java.getResourceAsStream("/airports.csv")?.let { InputStreamReader(it) })
            .readAll()
    }
}